projects/core/bindaas-trusted-application-manager/src/main/java/edu/emory/cci/bindaas/trusted_app/impl/TrustedApplicationManagerImpl.java

```diff
@@ -399,125 +439,155 @@ else if (bindaasConfigurationProtocol.equals("JWT") && protocol.equals("jwt")) {
+						: comments;
+				Integer count = apiKeyManager.revokeAPIKey(
+						new BindaasUser(username), trustedAppEntry.getName(),
+						comments, ActivityType.REVOKE);
 
-			JsonObject retVal = new JsonObject();
-			retVal.add("username", new JsonPrimitive(username));
-			retVal.add("keys_deleted", new JsonPrimitive(count));
-			retVal.add("applicationID", new JsonPrimitive(applicationID));
+				JsonObject retVal = new JsonObject();
+				retVal.add("username", new JsonPrimitive(username));
+				retVal.add("keys_deleted", new JsonPrimitive(count));
+				retVal.add("applicationID", new JsonPrimitive(applicationID));
 
-			retVal.add("applicationName",
-					new JsonPrimitive(trustedAppEntry.getName()));
-			return Response.ok().entity(retVal.toString())
-					.type("application/json").build();
+				retVal.add("applicationName",
+						new JsonPrimitive(trustedAppEntry.getName()));
+				return Response.ok().entity(retVal.toString())
+						.type("application/json").build();
 
-		} catch (NotAuthorizedException e) {
-			return exceptionToResponse(e, applicationID, "not-resolved",
-					username);
-
-		} catch (APIKeyManagerException apiKeyManagerException) {
-			switch (apiKeyManagerException.getReason()) {
-			case KEY_ALREADY_EXIST:
-				return exceptionToResponse(new DuplicateAPIKeyException(
-						username), applicationID, "not-resolved", username);
-			case KEY_DOES_NOT_EXIST:
-				return exceptionToResponse(new APIKeyDoesNotExistException(
-						username), applicationID, "not-resolved", username);
-			default:
-				return Response.status(500)
-						.entity(apiKeyManagerException.getMessage()).build();
+			} catch (NotAuthorizedException e) {
+				return exceptionToResponse(e, applicationID, "not-resolved",
+						username);
+
+			} catch (Exception e) {
+				return exceptionToResponse(e.getMessage());
+			}
+		}
+		else if (bindaasConfigurationProtocol.equals("JWT") && protocol.equals("jwt")) {
+			try {
+				TrustedApplicationEntry trustedAppEntry = authenticateTrustedApplication(
+						applicationID, salt, digest, username);
+
+				comments = comments == null ? comments = "JWT revoked via Trusted Application API"
+						: comments;
+				Integer count = JWTManager.revokeJWT(new BindaasUser(username),
+						trustedAppEntry.getName(), comments, ActivityType.REVOKE);
+
+				JsonObject retVal = new JsonObject();
+				retVal.add("username", new JsonPrimitive(username));
+				retVal.add("tokens_deleted", new JsonPrimitive(count));
+				retVal.add("applicationID", new JsonPrimitive(applicationID));
+
+				retVal.add("applicationName",
+						new JsonPrimitive(trustedAppEntry.getName()));
+				return Response.ok().entity(retVal.toString())
+						.type("application/json").build();
+
+			} catch (NotAuthorizedException e) {
+				return exceptionToResponse(e, applicationID, "not-resolved",
+						username);
+
+			} catch (Exception e) {
+				return exceptionToResponse(e.getMessage());
```

Comment:
Instead of just returning the exceptions, try to add some helpful logs wherever appropriate.

For example:

instead of above:

} catch (Exception e) {
                                Strong errorMsg = "Very helpful message here"
                                log.error(errorMessage);
				return exceptionToResponse(errorMsg,  e);

Reply:
Of course, keep a balance. If you think a log.error is unnecessary, just append the additional error message before returning (throwing the error).

Reply:
There will be three kinds of exceptions:
1. Not Authorized
2. API/JWT Manager related
3. General

For 1&2 we are returning HTTP Exceptions which have error details and suggested action.
Since 3 is general I don't know what exactly to put as a "helpful" message.

In any case, logging is not being done anywhere. For 1&2 I can do
```
log.error("helpful message");
return exceptionToResponse(HTTPException, other args)
```

For 3 I can only do
```
log.error(e)
return exceptionToResponse(e);
```

Reply:
+1